controllers/categoryController.test.js

```diff
@@ -111,6 +113,128 @@ describe('createCategoryController', () => {
     });
 })
 
+describe('updateCategoryController', () => {
+    let req, res, consoleLogSpy;
+    const error = new Error('Database error');
+
+    beforeEach(() => {
+        req = {};
+        res = {
+            status: jest.fn().mockReturnThis(),
+            send: jest.fn(),
+        };
+    });
+
+    test('should return 200 with valid name and id', async () => {
+        
+        req.body = { name: 'test' }; 
+        req.params = { id: 1 };
+
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+        const updatedCategory = { name: 'test', id: 1, slug: 'test' };
+        categoryModel.findByIdAndUpdate = jest.fn().mockResolvedValue(updatedCategory);
+
+        await updateCategoryController(req, res);
+        
+        expect(res.status).toHaveBeenCalledWith(200);
+        expect(res.send).toHaveBeenCalledWith({
+            success: true,
+            messsage: "Category Updated Successfully",
+            category: updatedCategory
+        });
+    });
+
+    test('should return 500 and error while updating category without id', async () => {
+        consoleLogSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
+        req.body = { name: 'test' }; 
+        categoryModel.findOne = jest.fn().mockRejectedValue(error);
+
+        await updateCategoryController(req, res);
+        expect(res.status).toHaveBeenCalledWith(500);
+        expect(res.send).toHaveBeenCalledWith({
+            success: false,
+            error: expect.any(Error),
```

Comment:
For the `error`, why not expect the constant error variable that you have instantiated, instead of using `expect.any(Error)`?